### PR TITLE
[RAPPS] application-view refactor

### PR DESCRIPTION
**Pull request by kernelbin**
## Purpose

refactor application-view, fix some bugs and some trivial improvements

- application-view is now in a separate file (```appview.cpp```, corresponding header ```appview.h```). and add a header for ```gui.cpp``` (```gui.h```)

- ```GetSystemColorDepth``` are now moved into ```misc.cpp```
- fix missing ```inline``` in ```sdk/lib/atl/atltypes.h```
- fix & improve tab-order handling
- now double click on an item, or right click on an item and select Install in popup-menu will install exactly **this** application

## TODO

- [x] handle the situation that in uninstall registry, ```InstallDate``` might be a ```DWORD``` (seems like a UNIX time-stamp. investigation required)

### CODE REVIEW

File: base/applications/rapps/appview.cpp [hunk 1]
```diff
@@ -0,0 +1,1620 @@
...
+#include <shlobj_undoc.h>
+#include <shlguid_undoc.h>
+
+#include <atlbase.h>
+#include <atlcom.h>
+#include <atltypes.h>
+#include <atlwin.h>
+#include <wininet.h>
+#include <shellutils.h>
+#include <rosctrls.h>
```
Comment (learn-more) [inline base/applications/rapps/appview.cpp:24]: Doesnt rapps.h include most of this?

File: base/applications/rapps/appview.cpp [hunk 2]
```diff
@@ -0,0 +1,1620 @@
...
+#include <gdiplus.h>
+#include <math.h>
+
+
+class CMainWindow;
```
Comment (learn-more) [inline base/applications/rapps/appview.cpp:29]: Is this needed?

File: base/applications/rapps/appview.cpp [hunk 3]
```diff
@@ -0,0 +1,1620 @@
...
+        hColumn.fmt &= ~(HDF_SORTUP | HDF_SORTDOWN);
+        Header_SetItem(hHeader, nLastHeaderID, &hColumn);
+    }
+
+    /* Set the sorting style to the new column */
+    hColumn.mask = HDI_FORMAT;
+    Header_GetItem(hHeader, nHeaderID, &hColumn);
+
+    hColumn.fmt &= (bIsAscending ? ~HDF_SORTDOWN : ~HDF_SORTUP);
+    hColumn.fmt |= (bIsAscending ? HDF_SORTUP : HDF_SORTDOWN);
```
Comment (learn-more) [inline base/applications/rapps/appview.cpp:955]: Another option is:
```
   hColumn.fmt &= ~(HDF_SORTDOWN | HDF_SORTUP);
    hColumn.fmt |= (bIsAscending ? HDF_SORTUP : HDF_SORTDOWN);
```

File: base/applications/rapps/appview.cpp (hunk 1 above)
Comment (kernelbin) [reply base/applications/rapps/appview.cpp]: yes. I removed them

File: base/applications/rapps/appview.cpp (hunk 2 above)
Comment (kernelbin) [reply base/applications/rapps/appview.cpp]: no.... I deleted it now

File: base/applications/rapps/appview.cpp (hunk 3 above)
Comment (kernelbin) [reply base/applications/rapps/appview.cpp]: yes this seems easier to understand

**Pull request merged**